server/src/HttpServer.hs

```diff
@@ -16,29 +16,25 @@ import qualified Data.Text.Lazy as LText
 import qualified Network.Wai as Wai
 import qualified Web.Scotty.Trans as Scotty
 
-import qualified System.Log.Raven.Types as Sentry
-
 import JwtMiddleware (jwtMiddleware)
 import Core (Core (..), EnqueueResult (..))
 import Config (Config (..))
+import Logger
 import qualified Store
 import qualified Core
 import qualified Metrics
-import qualified SentryLogging
 
-new :: Maybe Sentry.SentryService -> Core -> IO Application
-new mSentryService core =
+new :: Core -> IO Application
+new core =
   scottyApp $ do
     -- first middleware is the outermost. this has to be the metrics middleware
     -- in order to intercept all requests their corresponding responses
     forM_ (coreMetrics core) $ middleware . metricsMiddleware
     -- Use the Sentry logger if available
     -- Scottys error handler will only catch errors that are thrown from within
     -- a ```liftAndCatchIO``` function.
-    if isJust mSentryService then
-        Scotty.defaultHandler (liftIO . SentryLogging.logCrashMessage "Thread handler error" mSentryService . show)
-    else
-        Scotty.defaultHandler (liftIO . putStrLn . LText.unpack )
+    Scotty.defaultHandler (liftIO . postLog (coreLogger core) LogError . pack . show)
```

Comment:
If I understand the documentation correctly, the default handler is also responsible for actually responding the HTTP request.
That can be a simple `503` with a plain text "Internal server error" body.